k2/csrc/fsa_algo.h

```diff
@@ -298,6 +298,38 @@ Ragged<int32_t> ShortestPath(FsaVec &fsas,
  */
 Fsa Union(FsaVec &fsas, Array1<int32_t> *arc_map = nullptr);
 
+/* Compute the closure of an FSA.
+
+   The closure is implemented in the following way:
+
+   - For all arcs entering the final-state, change the destination-state from
+   the final-state to state 0 (the start state), and the label from -1 to 0.
+
+   - Add an arc from the start-state to the final-state (with label -1, and
+   score 0, of course); this can be the last numbered arc of the start state of
+   the output. The implementation wouldn't be in-place, we'd create a new FSA.
+   Assume that the original start state has two arcs, then the arc-map would be
+   [ 0 1 -1 2 3 ... ].
+
+   Note: the output will be arc-sorted if the input was arc-sorted and
```

Comment:
How does the current implementation make sure the ouput arc-sorted if input arc-sorted? As we changed label from -1 to 0 for those final-arcs, accoding to our definition of arc-sort (compared labels with type `uint32_t` instead of `int32_t`), the output would not be arc-sorted as arc with label `0` now is the last arc leaving a state.

Reply:
The output fsa will not be arc sorted even if the input fsa was arc sorted. I am updating the documentation.

Reply:
Just metion it, actually we can do this by:
1. do `count` on the input Fsa to get how many epsilon-arcs we have for each state
2. for a specific arc, if it's label == 0, then just copy to the corresponding position in the output Fsa; if lable != 0, then copy to the corresponding position +1. 
3. Add an arc from this state to start state, with label == 0 at position count.